kukur/source/piwebapi_af/piwebapi_af.py

```diff
@@ -491,8 +492,7 @@ def read_data(
             params=params,
         )
         if response.status_code == NOT_FOUND:
-            logger.warning("No data found for %s", data_request.url)
-            return pa.Table.from_pydict({"ts": [], "value": [], "quality": []})
+            raise InvalidDataError(f"Data not found for {data_request.url}")
```

Comment:
The exception should probably be a new exception, so it can be acted upon. An `InvalidDataError` could be thrown for many other reasons than a series not existing.